src/main/scala/com/github.sstone/amqp/ChannelOwner.scala

```diff
@@ -51,10 +60,9 @@ object ChannelOwner {
           def handleFlow(active: Boolean): Unit = listener ! HandleFlow(active)
         }))
       }
-      case request@Publish(exchange, routingKey, body, properties, mandatory, immediate) => {
+      case request@Publish(_, _, _, _, _, _) => {
```

Comment:
The bytecode is the same but the ASTs differ. Because of these differences we can detect pattern matches that violate parametricity in a lot of situations. Consider the following two functions:
```scala
   def foo[A](a : A) : Boolean = a match {
      case _ : Option[_] ⇒ true
      case _             ⇒ false
   }

   def bar[A](a : A) : Boolean = a match {
      case a @ Some(_) ⇒ true
      case _           ⇒ false
   }
```
Both are bad but with wartremover's Any wart the latter throws this error:
```console
[error] /home/chris/Programming/ArrXml/src/main/scala/arrxml/package.scala:21: Inferred type containing Any
[error]       case a @ Some(_) ⇒ 1
[error]                    ^
[error] one error found
[error] (compile:compile) Compilation failed
```

This particular example only applies to polymorphic deconstructors, but hopefully we can find a way to detect bad pattern matches which involve non-polymorphic deconstructors.